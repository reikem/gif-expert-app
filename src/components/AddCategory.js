import React, { useState } from "react";

const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = useState();

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("Submit echo");
    if (inputValue.trim().length > 2) {
      setCategory((cats) => [...cats, inputValue]);
      setCategory("");
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <h3>{inputValue} </h3>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  );
};

export default AddCategory;
