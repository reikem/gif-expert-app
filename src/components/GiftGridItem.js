import React from "react";
import PropTypes from "prop-types";
export const GiftGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__backInLeft">
      <img src={url} alt={title} />
      <p key={id}>{title}</p>
    </div>
  );
};

GiftGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
