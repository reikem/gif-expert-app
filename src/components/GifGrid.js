import React from "react";
import { useFetchGigs } from "../hooks/useFetchGifs";
import { GiftGridItem } from "./GiftGridItem";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGigs(category);

  return (
    <>
      <h3 className="animate__animated animate__bounceInDown">{category}</h3>
      {loading && <p className="animate__animated animate__flash">Loading</p>}
      {
        <div>
          <ol className="card-grid">
            {data.map((img) => (
              <GiftGridItem key={img.id} {...img} />
            ))}
          </ol>
        </div>
      }
    </>
  );
};
