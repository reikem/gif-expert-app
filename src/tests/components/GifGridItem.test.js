import React from "react";
import "../../setupTests";
import { shallow } from "enzyme";
import { GiftGridItem } from "../../components/GiftGridItem";

describe("Prueba componente GifGridItem", () => {
  const title = "un titulo";
  const url = "https://localhost/algo.jpg";
  const id = "12345";
  const wrapper = shallow(<GiftGridItem title={title} id={id} url={url} />);

  test("debe mostrar una imagen correcta del componente GifGridItem", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener la imagen igual a la url y alt de los props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener la animacion  fade in", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__backInLeft")).toBe(true);
  });
});
