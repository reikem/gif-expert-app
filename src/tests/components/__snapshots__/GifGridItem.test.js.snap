// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente GifGridItem debe mostrar una imagen correcta del componente GifGridItem 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p
    key="12345"
  >
    un titulo
  </p>
</div>
`;
